cmd/ping/main.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// binary for the pinger service for RTT measurement.
+package main
+
+import (
+	"context"
+	"net/http"
+	"strings"
+	"time"
+
+	"agones.dev/agones/pkg"
+	"agones.dev/agones/pkg/util/runtime"
+	"agones.dev/agones/pkg/util/signals"
+	"github.com/pkg/errors"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+	"golang.org/x/time/rate"
+)
+
+const (
+	httpResponseFlag = "http-response"
+	udpRateLimitFlag = "udp-rate-limit"
+)
+
+var (
+	logger = runtime.NewLoggerWithSource("main")
+)
+
+func main() {
+	ctlConf := parseEnvFlags()
+	if err := ctlConf.validate(); err != nil {
+		logger.WithError(err).Fatal("could not create controller from environment or flags")
+	}
+
+	logger.WithField("version", pkg.Version).
+		WithField("ctlConf", ctlConf).Info("starting ping...")
+
+	stop := signals.NewStopChannel()
+
+	udpSrv := serveUDP(ctlConf, stop)
+	defer udpSrv.close()
+
+	cancel := serveHTTP(ctlConf)
+	defer cancel()
+
+	<-stop
+	logger.Info("shutting down...")
+}
+
+func serveUDP(ctlConf config, stop <-chan struct{}) *udpServer {
+	s := newUDPServer(ctlConf.UDPRateLimit)
+	s.run(stop)
+	return s
+}
+
+// serveHTTP starts the HTTP handler, and returns a cancel/shutdown function
+func serveHTTP(ctlConf config) func() {
+	// we don't need a health checker, we already have a http endpoint that returns 200
+	mux := http.NewServeMux()
+	srv := &http.Server{
+		Addr:    ":8080",
+		Handler: mux,
+	}
+
+	mux.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
```

Comment:
should you include the healthcheck library from heptio like we did in the agones controller ? I'm saying this because I have a feeling that currently if the udp listen fail the container is still consider healthy, however you could have register an healhtcheck from the udp listener.

This is not a strongly held opinion, just some thoughts.

Reply:
That's a good thought. I will review and see how I can check the health of the udp listener. :+1:

Reply:
Done! Nice call!